### PR TITLE
Part of 3 of beacon state Altair pkg - copy and hash tree root 

### PR DESCRIPTION
**Pull request by terencechain**
Follow up on #8658 

Add copy, hash tree root. Most of them are identical to existing ones. Will add commentary in code

Note: Some tests are missing and that will come in next part (For preview of tests, check out #8584). The reason that we can't do tests in this PR is because tests require the testutil functions. If we include them then the diff will be 3000+

### CODE REVIEW

File: beacon-chain/state/state-altair/state_trie.go [hunk 1]
```diff
@@ -59,3 +68,327 @@ func InitializeFromProtoUnsafe(st *pbp2p.BeaconStateAltair) (*BeaconState, error
...
+		return participationBitsRoot(b.state.CurrentEpochParticipation)
+	case justificationBits:
+		return bytesutil.ToBytes32(b.state.JustificationBits), nil
+	case previousJustifiedCheckpoint:
+		return htrutils.CheckpointRoot(hasher, b.state.PreviousJustifiedCheckpoint)
+	case currentJustifiedCheckpoint:
+		return htrutils.CheckpointRoot(hasher, b.state.CurrentJustifiedCheckpoint)
+	case finalizedCheckpoint:
+		return htrutils.CheckpointRoot(hasher, b.state.FinalizedCheckpoint)
+	case inactivityScores:
```
Comment (terencechain) [inline beacon-chain/state/state-altair/state_trie.go:352]: L352-L356 are now

File: beacon-chain/state/state-altair/state_trie.go [hunk 2]
```diff
@@ -59,3 +68,327 @@ func InitializeFromProtoUnsafe(st *pbp2p.BeaconStateAltair) (*BeaconState, error
...
+			// Everything else, too small to be concerned about, constant size.
+			Fork:                        b.fork(),
+			LatestBlockHeader:           b.latestBlockHeader(),
+			Eth1Data:                    b.eth1Data(),
+			JustificationBits:           b.justificationBits(),
+			PreviousJustifiedCheckpoint: b.previousJustifiedCheckpoint(),
+			CurrentJustifiedCheckpoint:  b.currentJustifiedCheckpoint(),
+			FinalizedCheckpoint:         b.finalizedCheckpoint(),
+			GenesisValidatorsRoot:       b.genesisValidatorRoot(),
+			CurrentSyncCommittee:        b.CurrentSyncCommittee(),
```
Comment (terencechain) [inline beacon-chain/state/state-altair/state_trie.go:112]: New line

File: beacon-chain/state/state-altair/state_trie.go [hunk 3]
```diff
@@ -59,3 +68,327 @@ func InitializeFromProtoUnsafe(st *pbp2p.BeaconStateAltair) (*BeaconState, error
...
+			Slashings:     b.state.Slashings,
+			Eth1DataVotes: b.state.Eth1DataVotes,
+
+			// Large arrays, increases over time.
+			Validators:                 b.state.Validators,
+			Balances:                   b.state.Balances,
+			HistoricalRoots:            b.state.HistoricalRoots,
+			PreviousEpochParticipation: b.state.PreviousEpochParticipation,
+			CurrentEpochParticipation:  b.state.CurrentEpochParticipation,
+			InactivityScores:           b.state.InactivityScores,
```
Comment (terencechain) [inline beacon-chain/state/state-altair/state_trie.go:101]: New line

File: beacon-chain/state/state-altair/deprecated_setters.go [hunk 4]
```diff
@@ -24,3 +25,8 @@ func (b *BeaconState) AppendCurrentEpochAttestations(val *pbp2p.PendingAttestati
 func (b *BeaconState) AppendPreviousEpochAttestations(val *pbp2p.PendingAttestation) error {
 	return errors.New("AppendPreviousEpochAttestations is not supported for hard fork 1 beacon state")
 }
+
+// ToProto is not supported for HF1 beacon state.
+func (b *BeaconState) ToProto() (*v1.BeaconState, error) {
```
Comment (terencechain) [inline beacon-chain/state/state-altair/deprecated_setters.go:30]: Blocked. Support this later

**Review by rauljordan — APPROVED** [review]

**Pull request merged**